### PR TITLE
Fix config style for new Supervisor/Hardware

### PR DESCRIPTION
**Pull request by TheZoker**
Fixes: https://github.com/esphome/issues/issues/1806

Since version `2021.02.0` of the home assistant supervisor, there is a new way of handling hardware:
https://github.com/home-assistant/supervisor/pull/2429

This PR is intended to update the config files to the latest requirements.

Here is an example, how the addons from the core repository are fixed:
https://github.com/home-assistant/addons/pull/1825

### CODE REVIEW

**Comment by pvizeli** [comment]
The new supervisor is also plug&play ;)

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        with:
+          path: "./esphome"
+      - name: 🚀 Run Home Assistant Add-on Lint on ESPHome-Beta
+        uses: frenck/action-addon-linter@v2
+        with:
+          path: "./esphome-beta"
+      - name: 🚀 Run Home Assistant Add-on Lint on ESPHome-Dev
+        uses: frenck/action-addon-linter@v2
+        with:
+          path: "./esphome-dev"
```
Comment (frenck) [inline .github/workflows/lint.yml:21]:
```suggestion
    strategy:
       matrix:
         channel: [esphome, esphome-beta, esphome-dev]
    steps:
      - name: ⤵️ Check out code from GitHub
        uses: actions/checkout@v2
      - name: 🚀 Run Home Assistant Add-on Lint on ESPHome
        uses: frenck/action-addon-linter@v2
        with:
          path: "./${{ matrix.channel }}"
```

**Review by frenck — APPROVED** [review]
Besides a small suggestion on the GitHub Action, it looks good to me 👍

**Comment by glmnet** [comment]
Thanks, does this need a version bump?

**Comment by jesserockz** [comment]
These changes do not hold as the version bump script regenerates the json from the yaml file in the template folder..

**Pull request merged**